Documentation/tutorial/walkthrough_create_a_docfx_project.md

```diff
@@ -0,0 +1,183 @@
+Walkthrough: Create a docfx project
+===================================
+
+By completing this walkthrough, you'll become familiar with the workflow of `docfx`, the general principle of organizing documents inside `docfx` and most of the functionality inside `docfx` to generate a documentation website with your own design.
+
+Step1. Setup docfx
+------------------------
+Download *docfx* from http://dotnet.github.io/docfx/. [Getting Started with docfx](http://dotnet.github.io/docfx/tutorial/docfx_getting_started.html#2-use-docfx-exe-directly) describes 3 ways to install *docfx*. This walkthrough uses the first one: use *docfx.exe* directly.
+
+1. Download *docfx.zip* and unzip it to `D:\docfx\`
+2. Add `D:\docfx\` to `PATH` so that command `docfx` and be directly called from everywhere for convenience.
+
+Step2. Init a docfx project
+---------------------------
+1. Create a new folder `D:\docfx_walkthrough`  
+2. Start Command Line under `D:\docfx_walkthrough`
+3. Call `docfx init -q`. This command generates a `docfx.json` file under the same folder. `docfx.json` is the configuration file `docfx` uses to generate documentation. `-q` option means generating `docfx.json` quitely using default value, you can also try `docfx init` and follow the instructions to provide your own settings. The content of `docfx.json` generated is as below. It contains 2 parts, one is `metadata` part and another is `build` part. We will go through each part in detail in the following steps.
+
+```json
+{
+  "metadata": [
+    {
+      "src": [
+        {
+          "files": [ "src/**.csproj" ],
+          "exclude": [ "**/bin/**", "**/obj/**" ]
+        }
+      ],
+      "dest": "obj/api"
+    }
+  ],
+  "build": {
+    "content": [
+      {
+        "files": [ "api/**.yml" ],
+        "src": "obj"
+      },
+      {
+        "files": [ "articles/**/*.md", "toc.yml", "*.md" ],
+        "exclude": [ "**/bin/**", "**/obj/**" ]
+      }
+    ],
+    "resource": [
+      {
+        "files": [ "images/**" ],
+        "exclude": [ "**/bin/**", "**/obj/**" ]
+      }
+    ],
+    "overwrite": [
+      {
+        "files": [ "apidoc/**.md" ],
+        "exclude": [ "**/bin/**", "**/obj/**" ]
+      }
+    ],
+    "dest": "_site",
+    "template": [ "default" ]
+  }
+}
+``` 
+
+Step3. `metadata` part: generate metadata for a c# project
+---------------------------
+1. Create a subfolder `src` under `D:\docfx_walkthrough`. Open *Visual Studio Community 2015* or above and create a **C# Class Library** `HelloDocfx` under folder `src`. In the `Class1.cs`, add some comments and methods to this class, as similar to:
```

Comment:
Add summary for this step? 
In this step,  we will extract metadata from C# source code (`*.cs`) into YAML files (`.yml`). These `yml` files is mainly used to generate reference documentation in *Step4*.